<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ihm_ModuleDesigner</class>
 <widget class="QMainWindow" name="ihm_ModuleDesigner">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>489</width>
    <height>356</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="statusTip">
       <string/>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Création module</string>
       </attribute>
       <attribute name="toolTip">
        <string>Créer un nouveau module</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Nom du module</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QRadioButton" name="ModuleTypePlugin">
          <property name="text">
           <string>Plugin Module</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QRadioButton" name="ModuleTypeBasic">
          <property name="text">
           <string>Basic Module</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="2">
         <widget class="QLineEdit" name="auteur">
          <property name="text">
           <string>?</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="2">
         <widget class="QLineEdit" name="nom_module">
          <property name="text">
           <string>?</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2" rowspan="2">
         <widget class="QPushButton" name="PB_generate">
          <property name="toolTip">
           <string>Crée un répertoire et génère les fichiers dedans</string>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(0, 255, 127);
</string>
          </property>
          <property name="text">
           <string>Créer le module</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0" colspan="2">
         <widget class="QCheckBox" name="chbx_integrer_modume_au_projet">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Modifie les fichiers .pro / .cpp / .h du projet pour intégrer automatiquement le nouveau module. Décocher cette case pour que le module soit créé mais que les fichiers du projets ne soient pas modifiés.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Intégrer le module généré au projet</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Auteur</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QToolButton" name="PB_choix_repertoire">
          <property name="text">
           <string>...</string>
          </property>
          <property name="popupMode">
           <enum>QToolButton::DelayedPopup</enum>
          </property>
          <property name="autoRaise">
           <bool>false</bool>
          </property>
          <property name="arrowType">
           <enum>Qt::NoArrow</enum>
          </property>
         </widget>
        </item>
        <item row="7" column="0" rowspan="2" colspan="4">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="1" colspan="2">
         <widget class="QLineEdit" name="repertoire_projet">
          <property name="toolTip">
           <string>Indique le répertoire du projet LaBotBox (répertoire où se trouve le fichier .pro)</string>
          </property>
          <property name="text">
           <string>./</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QLineEdit" name="description">
          <property name="text">
           <string>?</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Description</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QCheckBox" name="chbx_interface_graphique">
          <property name="toolTip">
           <string>Indique si le module est avec ou sans IHM</string>
          </property>
          <property name="text">
           <string>Inteface graphique</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Répertoire du projet</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QCheckBox" name="chbx_guide_utilisation">
          <property name="text">
           <string>Guide d'utilisation</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Intégration module</string>
       </attribute>
       <attribute name="toolTip">
        <string>Intégrer au projet un module déjà existant</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Emplacement  du module</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="repertoire_module_existant">
          <property name="toolTip">
           <string>Empalcement du module dans le projet</string>
          </property>
          <property name="text">
           <string>./</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QToolButton" name="PB_choix_repertoire_module_existant">
          <property name="text">
           <string>...</string>
          </property>
          <property name="popupMode">
           <enum>QToolButton::DelayedPopup</enum>
          </property>
          <property name="autoRaise">
           <bool>false</bool>
          </property>
          <property name="arrowType">
           <enum>Qt::NoArrow</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="PB_integrer_module_existant">
          <property name="toolTip">
           <string>Crée un répertoire et génère les fichiers dedans</string>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(0, 255, 127);
</string>
          </property>
          <property name="text">
           <string>Intégrer le module</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;INFO&lt;/span&gt;: permet d'intégrer au projet un module tout fait (livré par quelqu'un d'autre par exemple).&lt;/p&gt;&lt;p&gt;Le module doit être préalablement copié dans le répertoire BasicModule ou PluginModule suivant son type.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Désintégration module</string>
       </attribute>
       <attribute name="toolTip">
        <string>Désintégrer du projet un module</string>
       </attribute>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>140</y>
          <width>181</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Emplacement  du module</string>
        </property>
       </widget>
       <widget class="QPushButton" name="PB_desintegrer_module">
        <property name="geometry">
         <rect>
          <x>137</x>
          <y>166</y>
          <width>177</width>
          <height>23</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Crée un répertoire et génère les fichiers dedans</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 0, 0);
</string>
        </property>
        <property name="text">
         <string>Désintégrer le module</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>13</x>
          <y>5</y>
          <width>332</width>
          <height>129</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;INFO&lt;/span&gt;: permet de supprimer du projet les références à un module. &lt;/p&gt;&lt;p&gt;Les fichiers et le répertoire du module ne sont PAS supprimés.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLineEdit" name="repertoire_module_desintegration">
        <property name="geometry">
         <rect>
          <x>190</x>
          <y>140</y>
          <width>177</width>
          <height>20</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Empalcement du module dans le projet</string>
        </property>
        <property name="text">
         <string>./</string>
        </property>
       </widget>
       <widget class="QToolButton" name="PB_choix_repertoire_module_desintegration">
        <property name="geometry">
         <rect>
          <x>370</x>
          <y>140</y>
          <width>25</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="popupMode">
         <enum>QToolButton::DelayedPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>false</bool>
        </property>
        <property name="arrowType">
         <enum>Qt::NoArrow</enum>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>489</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
